src/core/i18n/translate.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react';
+
+
+function getDisplayName(component) {
+  return component.displayName || component.name || 'Component';
+}
+
+
+export default function translate(options = {}) {
+  const { withRef = false } = options;
+
+  return function Wrapper(WrappedComponent) {
+    class Translate extends Component {
+      constructor(props, context) {
+        super(props, context);
+        this.i18n = context.i18n;
+        this.state = {
+          i18nLoadedAt: null,
+        };
+      }
+
+      getChildContext() {
+        return { i18n: this.i18n };
```

Comment:
Is this needed? I feel like it should already be passed down since it is being received in `context`.